packages/form-upload/Upload.js

```diff
@@ -58,22 +38,39 @@ const Upload = ({
 
   const fieldValue = Array.isArray(field.value) ? field.value : [];
 
+  const fileDeliveryProps =
+    rest.deliveryChannel && rest.metadata
+      ? {
+          deliveries: [
+            {
+              deliveryChannel: rest.deliveryChannel,
+              metadata: rest.metadata,
+            },
+          ],
+        }
+      : undefined;
+
+  const callFileDelivery = (data, config, upload) => {
+    if (!Array.isArray(upload)) upload = [upload];
+    upload.forEach(upload => {
+      data.deliveries[0].fileURI = upload.references[0];
+      try {
+        avFilesDeliveryApi.uploadFilesDelivery(data, config);
+      } catch (error) {
+        setFieldError(error);
+      }
+    });
+  };
+
   useEffect(() => {
-    if (rest.fileDeliveryOnSubmit && isSubmitting === true) {
-      fileDeliveryOnSubmit(
-        rest.fileDeliveryProps,
+    if (rest.deliverFileOnSubmit && isSubmitting === true) {
```

Comment:
Formik has a few [phases to form submission](https://formik.org/docs/guides/form-submission#submission-phases) that we'll need to check. Mainly we want to make sure that the submission handler is finished validating, there are no errors, and the submission handler is running. That still may not be foolproof but I'll keep looking in to it